### ISSUE TITLE
app crash when migrate protobuf from protobuf-java to protobuf-javalite

### ISSUE THREAD

**Issue by DysaniazzZ** [protocolbuffers/protobuf#6565]
**What version of protobuf and what language are you using?**
Version: 3.9.1
Language:Java

**What operating system (Linux, Windows, ...) and version?**
Android, API 23

my configs are below:
root build.gradle:
```
buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.4.2'
        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.10'
    }
}

...
```

module build.gradle:

```
...

protobuf {
    protoc {
        artifact = 'com.google.protobuf:protoc:3.9.1'
    }
    generateProtoTasks {
        all().each { task ->
            task.builtins {
                java {
                    option "lite"
                }
            }
        }
    }
}

dependencies {
    api 'com.google.protobuf:protobuf-javalite:3.9.1'
}
```

but the app crash when I open it, and the crash logs are below:

```
java.lang.NoSuchMethodError: No virtual method putBoolean(Ljava/lang/Object;JZ)V in class Lsun/misc/Unsafe; or its super classes (declaration of 'sun.misc.Unsafe' appears in /system/framework/core-libart.jar)
        at com.google.protobuf.UnsafeUtil$JvmMemoryAccessor.putBoolean(UnsafeUtil.java:620)
        at com.google.protobuf.UnsafeUtil.putBoolean(UnsafeUtil.java:143)
        at com.google.protobuf.MessageSchema.parseProto3Message(MessageSchema.java:5255)
        at com.google.protobuf.MessageSchema.mergeFrom(MessageSchema.java:5396)
        at com.google.protobuf.ArrayDecoders.decodeMessageField(ArrayDecoders.java:246)
        at com.google.protobuf.ArrayDecoders.decodeMessageList(ArrayDecoders.java:704)
        at com.google.protobuf.MessageSchema.parseProto3Message(MessageSchema.java:5330)
        at com.google.protobuf.MessageSchema.mergeFrom(MessageSchema.java:5396)
        at com.google.protobuf.ArrayDecoders.decodeMessageField(ArrayDecoders.java:246)
        at com.google.protobuf.ArrayDecoders.decodeMessageList(ArrayDecoders.java:704)
        at com.google.protobuf.MessageSchema.parseProto3Message(MessageSchema.java:5330)
        at com.google.protobuf.MessageSchema.mergeFrom(MessageSchema.java:5396)
        at com.google.protobuf.GeneratedMessageLite.parsePartialFrom(GeneratedMessageLite.java:1552)
        at com.google.protobuf.GeneratedMessageLite.parseFrom(GeneratedMessageLite.java:1652)
        at xxx.entity.ChannelGroup$ChannelGroupProto.parseFrom(ChannelGroup.java:279)
```

**Comment by grill2010** [comment]
I think this is the same issue as [mine](https://github.com/protocolbuffers/protobuf/issues/6551). Should be resolved in the next version.

**Comment by DysaniazzZ** [comment]
@grill2010 Yes, I think so. Look forward to the next release.

**Comment by rafi-kamal** [comment]
This is a duplicate of https://github.com/protocolbuffers/protobuf/issues/6551, which was fixed in 3.10 release. Please reopen the issue if the problem still exists in 3.10+.
